### PR TITLE
Update v1.0.1

### PR DESCRIPTION
**Pull request by sumeetrajpal**
*sendData() now accepts a array of payloads.
*Posts an array to Conctr vs singular payloads.
*Sanity check on timestamps in the agent.
*Updated version number.

### CODE REVIEW

File: conctr.agent.class.nut [hunk 1]
```diff
@@ -117,6 +118,9 @@ class Conctr {
                 // Set the time stamp if not set already
                 if (!("_ts" in v) || (v._ts == null)) {
                     v._ts <- time();
+                }else if(("_ts" in v) && typeof v._ts == "number" && ((v._ts.tostring().len() <= 10) ? v._ts < MIN_TIME : v._ts < MIN_TIME*1000)){
```
Comment (blindman2k) [inline conctr.agent.class.nut]: Get the formatting/spacing right - space before `else` and after `if`.
Put brackets around the middle `==` statement.
The third statement there needs to be simplified or moved out into a variable. It's way too complex.
The log should say "Setting to time()".

File: conctr.device.class.nut [hunk 2]
```diff
@@ -94,39 +94,59 @@ class Conctr {
...
+                    // Store the callback for later
+                    if (callback) _onResponse[v._id] <- callback;
+
+                    //all payloads have been processed, send array to agent
+                    if (k == payloadLength-1) {
 
-            // Todo: Add optional Bullwinkle here
-            // Store the callback for later
-            if (callback) _onResponse[payload._id] <- callback;
+                        if(_DEBUG){
```
Comment (blindman2k) [inline conctr.device.class.nut]: Please correct all the `if(` and `}else` typos.

File: conctr.device.class.nut [hunk 3]
```diff
@@ -94,39 +94,59 @@ class Conctr {
...
-            // Store the location (wifis) if we have it and if it's not already set
-            if ((wifis != null) && !("_location" in payload)) {
-                payload._location <- wifis;
-            }
+                // Add an unique id for tracking the response
+                v._id <- format("%d:%d", hardware.millis(), hardware.micros());
+                v._source <- SOURCE_DEVICE;
+
+                // Todo: Don't getWifis if the _location is already set
+                _getWifis(function(wifis) {
```
Comment (blindman2k) [inline conctr.device.class.nut]: Do you really want to call `_getWifis()` for every loops? I doubt it.

**Review by blindman2k — CHANGES_REQUESTED** [review]

**Comment by sumeetrajpal** [comment]
Made requested changes.

**Comment by blindman2k** [comment]
I submitted a pull request to your repo sumeetrajpal. Merge that and then
merge that into this pull request.

**Review by blindman2k — APPROVED** [review]
I think we are finally there!

**Pull request merged**